### PR TITLE
Deprecate runtime guard envoy_reloadable_features_lua_respond_with_se…

### PR DESCRIPTION
**Pull request by sgovinda-ub**
…nd_local_reply

Signed-off-by: Shyam Govindaraj <shyirtt@gmail.com>

[Fixes #Issue]  https://github.com/envoyproxy/envoy/issues/25017

### CODE REVIEW

File: test/extensions/filters/http/lua/lua_filter_test.cc [hunk 1]
```diff
@@ -1284,12 +1284,9 @@ TEST_F(LuaHttpFilterTest, HttpCallNoBody) {
   callbacks->onSuccess(request, std::move(response_message));
   EXPECT_EQ(0, stats_store_.counter("test.lua.errors").value());
 }
-
+/*
```
Comment (sgovinda-ub) [inline test/extensions/filters/http/lua/lua_filter_test.cc:1287]: Need some help in fixing this test case.

**Comment by KBaichoo** [comment]
/assign @wbpcode 

@sgovinda-ub  I think you need to merge main for deps failure to be fixed, fix merge conflict

/wait

File: test/extensions/filters/http/lua/lua_filter_test.cc (hunk 1 above)
Comment (wbpcode) [reply test/extensions/filters/http/lua/lua_filter_test.cc:1287]: Maybe we just need to update the `encodeHeaders_` mock to `sendLocalReply` mock just you did in the `ImmediateResponse` case.

**Comment by sgovinda-ub** [comment]
@wbpcode  Please review the the updated PR.

File: test/extensions/filters/http/lua/lua_filter_test.cc [hunk 2]
```diff
@@ -1337,7 +1333,6 @@ TEST_F(LuaHttpFilterTest, HttpCallImmediateResponse) {
   Http::TestResponseHeaderMapImpl expected_headers{{":status", "403"},
                                                    {"set-cookie", "flavor=chocolate; Path=/"},
                                                    {"set-cookie", "variant=chewy; Path=/"}};
-  EXPECT_CALL(decoder_callbacks_, encodeHeaders_(HeaderMapEqualRef(&expected_headers), true));
```
Comment (wbpcode) [inline test/extensions/filters/http/lua/lua_filter_test.cc:1340]: The `expected_headers` is not be verified for now. Should we add a sendLocalReply mock here or remote this local variable?

**Review by wbpcode — COMMENTED** [review]
LGTM overall. Thanks. Only one comment.

File: test/extensions/filters/http/lua/lua_filter_test.cc (hunk 2 above)
Comment (sgovinda-ub) [reply test/extensions/filters/http/lua/lua_filter_test.cc:1340]: @wbpcode  sendLocalReply is being tested in other testcases. Can I remove the local variable for this case?

Comment (wbpcode) [reply test/extensions/filters/http/lua/lua_filter_test.cc:1340]: > @wbpcode sendLocalReply is being tested in other testcases. Can I remove the local variable for this case?

Seems like this case is covered by the `ImmediateResponse`, right? If is, we can remove this case directly.

Comment (sgovinda-ub) [reply test/extensions/filters/http/lua/lua_filter_test.cc:1340]: I had tried it earlier but failed with below error.
```
`Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
test/extensions/filters/http/lua/lua_filter_test.cc:1317:3: error: no member named 'gmock_sendLocalReply' in 'testing::NiceMock<Envoy::Http::MockAsyncClient>'
  EXPECT_CALL(cluster_manager_.thread_local_cluster_.async_client_, sendLocalReply(_, _, _, _, _))
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:2028:3: note: expanded from macro 'EXPECT_CALL'
  GMOCK_ON_CALL_IMPL_(obj, InternalExpectedAt, call)
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:2020:16: note: expanded from macro 'GMOCK_ON_CALL_IMPL_'
  ((mock_expr).gmock_##call)(::testing::internal::GetWithoutMatchers(), \
   ~~~~~~~~~~~ ^
<scratch space>:51:1: note: expanded from here
gmock_sendLocalReply`
```

**Comment by phlax** [comment]
@wbpcode this looks ready for another review

/wait-any

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**